python/lsst/ip/diffim/zogy.py

```diff
@@ -0,0 +1,839 @@
+        - doScorr : bool
+              Compute and return the corrected likelihood image S_corr
+              rather than the proper image difference
+        - inImageSpace : bool
+              Perform all convolutions in real (image) space rather than
+              in Fourier space. This option currently leads to artifacts
+              when using real (measured and noisy) PSFs, thus it is set
+              to `False` by default.
+            These kwargs may also include arguments to be propagated to
+            `ZogyTask.computeDiffim` and `ZogyTask.computeScorr`.
+
+        Returns
+        -------
+        A `lsst.pipe.base.Struct` containing the result of the
+        `subExposure` processing, labelled 'subExposure'. In this case
+        it is either the subExposure of the proper image difference D,
+        or (if `doScorr==True`) the corrected likelihood exposure `S`.
+
+        Notes
+        -----
+        This `run` method accepts parameters identical to those of
+        `ImageMapperSubtask.run`, since it is called from the
+        `ImageMapperTask`. See that class for more information.
+        """
+        bbox = subExposure.getBBox()
+        center = ((bbox.getBeginX() + bbox.getEndX()) // 2., (bbox.getBeginY() + bbox.getEndY()) // 2.)
+        center = afwGeom.Point2D(center[0], center[1])
+
+        imageSpace = kwargs.pop('inImageSpace', False)
+        doScorr = kwargs.pop('doScorr', False)
+        sigmas = kwargs.pop('sigmas', None)
+        padSize = kwargs.pop('padSize', 7)
+
+        # Psf and image for science img (index 2)
+        subExp2 = expandedSubExposure
+
+        # Psf and image for template img (index 1)
+        subExp1 = template.Factory(template, expandedSubExposure.getBBox())
+
+        if sigmas is None:
+            sig1 = sig2 = None
+        else:  # for testing, can use the input sigma (e.g., global value for entire exposure)
+            sig1, sig2 = sigmas[0], sigmas[1]
+
+        psf2 = subExp2.getPsf().computeImage(center).getArray()
+        if psf2.shape[0] < psf2.shape[1]:  # Sometimes CoaddPsf does this. Make it square.
+            psf2 = np.pad(psf2, ((1, 1), (0, 0)), mode='constant')
+        elif psf2.shape[0] > psf2.shape[1]:
+            psf2 = np.pad(psf2, ((0, 0), (1, 1)), mode='constant')
+
+        psf1 = template.getPsf().computeImage(center).getArray()
+        if psf1.shape[0] < psf1.shape[1]:  # Sometimes CoaddPsf does this. Make it square.
+            psf1 = np.pad(psf1, ((1, 1), (0, 0)), mode='constant')
+        elif psf1.shape[0] > psf1.shape[1]:
+            psf1 = np.pad(psf1, ((0, 0), (1, 1)), mode='constant')
+
+        # from diffimTests.diffimTests ...
+        if subExp1.getDimensions()[0] < psf1.shape[0] or subExp1.getDimensions()[1] < psf1.shape[1]:
+            return pipeBase.Struct(subExposure=subExposure)
+
+        psf1b = psf2b = None
+        if True and psf1.shape[0] == 41:  # it's a measured psf (hack!) Note this *really* helps for measured psfs.
```

Comment:
Since you're using this code again, it *really* should be a separate `kernelFilter` function. In that function, the size of the zero-padding should be configurable, but could have a default of 10 as you use here.
I don't think testing against the size of the psf equalling 41 is safe. Measured psfs will have a wide range of sizes.